### DISCUSSION TITLE
Hide Warning about Automatic Static Optimization

### DISCUSSION THREAD

**Discussion by wuifdesign** [vercel/next.js#35938] [Help] [answered]
Is there a way to hide the warning about:

```
Warning: You have opted-out of Automatic Static Optimization due to `getInitialProps` in `pages/_app`. This does not opt-out pages with `getStaticProps`
Read more: https://nextjs.org/docs/messages/opt-out-auto-static-optimization
```

I am aware of using this, and it is needed as i have to check the user and its config on every single page (this is why i fetch it on `App.getInitialProps`). I just want to remove the warning on build.

**Comment by icyJoseph** [comment, marked as answer]
As far as I know there's no way of silencing this warning. Here's the line where it happens in case that helps you, https://github.com/vercel/next.js/blob/4a7ab34baf932c102fbf281660f09f7eb98c4b74/packages/next/build/index.ts#L1141

**Reply by wuifdesign** [reply]
thank you. seams like you are right. maybe i should create an issue for this?

**Reply by icyJoseph** [reply]
Yeah, I don't know about that, personally I like to see all warnings clearly. I guess let people decide.

**Reply by uglow** [reply]
This affects people using the Rush.js monorepo tool. In their wisdom, they not-only halt the build process when a non-zero exit code occurs, but they *also* look for words like "Warning" and anything written to std-err. So being able to disable this warning would make life better for people that knowingly-opt-in to disabling the warning.

**Reply by 21echoes** [reply]
@uglow have you been able to work around this? I'm running into the same issue with my next.js app within a rush monorepo. I could turn on the rush flag `allowWarningsInSuccessfulBuild` but sadly that would apply to all projects, not just my one (only-internally-used) project that intentionally uses `_app` `getInitialProps`

**Reply by uglow** [reply]
@21echoes - yes, I did find a "workaround" by not using getInitialProps. Instead, include that code in a function which every page exports as `getStaticProps`.

**Reply by dhorkin** [reply]
you might be able to modify the solution for suppressing a totally unrelated warning here: https://github.com/wagmi-dev/viem/discussions/781#discussioncomment-6618771
